scripts/bin/update-packages

```diff
@@ -159,11 +159,8 @@ _fetch_and_cache_tags() {
 		for pkg_dir in "${repo_dir}"/*; do
 			quiet _should_update "${pkg_dir}" || continue # Skip if not needed.
 			grep -q '^termux_pkg_auto_update' "${pkg_dir}/build.sh" && continue # Skip if package has custom auto-update
-			if grep '^TERMUX_PKG_SRCURL=' "${pkg_dir}/build.sh" | grep -q 'github.com'; then
-				__GITHUB_PACKAGES+=("${pkg_dir}")
-			else
-				__PACKAGES+=("${pkg_dir}")
-			fi
+			grep -qE '^TERMUX_PKG_SRCURL=.*github\.com' "$pkg_dir/build.sh" && __GITHUB_PACKAGES+=("${pkg_dir}") || :
+			__PACKAGES+=("${pkg_dir}")
```

Comment:
Doesn't this cause duplicate entries?

Reply:
Why? It is effectively same to `grep '^TERMUX_PKG_SRCURL=' "${pkg_dir}/build.sh" | grep -q 'github.com'`.

Reply:
But spawns less processes.

Reply:
I would prefer to keep an if statement, as it is more readible

Reply:
Also this solution is a little bit faster.

Reply:
I mean the whole fetch&cache solution was designed because I used to invoke `update-packages` script on my pc with `PACKAGES_OPT_OUT=true` and it was very annoying to wait ~45 mins for script to finish.
Doing this as fast as possible is better both for my mental health and for our CI.

Reply:
> Why? It is effectively same to `grep '^TERMUX_PKG_SRCURL=' "${pkg_dir}/build.sh" | grep -q 'github.com'`.

Except that `__PACKAGES+=("${pkg_dir}")` runs in either case in the new version.

> Also this solution is a little bit faster.

I'm gonna need to you to show your work on that, because yes it might be slightly faster.
But if we're saving like 50ms on this part that's not worth the readability hit.

Reply:
> Except that `__PACKAGES+=("${pkg_dir}")` runs in either case in the new version.

Yes, it is intentional because I run over github packages list twice.

One time for extracting current versions, second time iterating over new versions and comparing them with current versions in order to drop/skip packages immediately (add them to seen list).

> I'm gonna need to you to show your work on that, because yes it might be slightly faster.
> But if we're saving like 50ms on this part that's not worth the readability hit.

50ms here, 50ms there and same thing in few more places and we have 500ms optimized per package. Optimizations are cumulative.

Probably we should run new version at least once in order to check how fast it will run.

Reply:
Agreed.

Reply:
I thought it is readible since we have hundreds regexps all over the code.

Reply:
I officially reduced the time of execution of this loop to 9 seconds (it is at least 29 seconds in original code) on my machine.

Also I reduced all `_should_update` invocation time by 6.5 times.

And yes, I added comments in all affected places.

Reply:
I was wrong. I measured time after dropping disk page caches.

Now the time is 6.8s compared to 24.2s.